### PR TITLE
Added DataTableGroupBy

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Added DataTableGroupBy

#### Where should the reviewer start?

DataTableGroupBy.js

#### What testing has been done on this PR?

added a story
added a simple unit test

#### How should this be manually tested?

story

#### Do Jest tests follow these best practices?

followed other Data tests for now

#### What are the relevant issues?

#6733 

#### Do the grommet docs need to be updated?

yes

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/Data/stories/Inline.js [hunk 1]
```diff
@@ -33,6 +34,7 @@ const Filters = ({ search, ...rest }) => (
   <DataFilters {...rest}>
     {search && <DataSearch property="sub.name" />}
     <DataFilter property="size" />
+    <DataTableGroupBy options={['size']} />
```
Comment (taysea) [inline src/js/components/Data/stories/Inline.js:37]: Can we add more data for this story? There's not really enough to make the grouping clear.

Also, can we add columns to the DataTable? It's a little confusing with no header cells.

Comment (jcfilben) [reply src/js/components/Data/stories/Inline.js:37]: ^ Agree, I think it is difficult to see group feature with such a small data set

Comment (jcfilben) [inline src/js/components/Data/stories/Inline.js:37]: In the case where there is only 1 item in each group it feels weird to show the expand/collapse control

https://user-images.githubusercontent.com/54560994/232119991-8f48b121-1438-4791-8d2e-2b111f279a09.mp4

File: src/js/components/DataTableGroupBy/stories/Simple.js
Comment (jcfilben) [inline src/js/components/DataTableGroupBy/stories/Simple.js:1]: This comment is outside the scope of this PR, but I'm wondering if we should revisit the styling for grouping in a DataTable. When I grouped by 'percent complete' in this example I felt that the grouping styling wasn't distinct enough to be helpful in visually parsing the data.

Screen Shot 2023-04-14 at 11 55 20 AM

File: src/js/components/Data/stories/Inline.js (hunk 1 above)
Comment (ericsoderberghp) [reply src/js/components/Data/stories/Inline.js:44]: Good thoughts. Added a table header and more rows.

[the remark this answers is not in the document]
Comment (ericsoderberghp) [reply src/js/components/Data/stories/Inline.js:44]: good catch! Fixed

Comment (ericsoderberghp) [reply src/js/components/Data/stories/Inline.js:44]: I've added a table header and more rows.

File: src/js/components/DataTableGroupBy/stories/Simple.js
Comment (ericsoderberghp) [reply src/js/components/DataTableGroupBy/stories/Simple.js:1]: Agreed. But, I'm thinking that should be a separate pull request.

File: src/js/components/DataTableGroupBy/propTypes.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+let PropType = {};
+if (process.env.NODE_ENV !== 'production') {
+  PropType = {};
```
Comment (jcfilben) [inline src/js/components/DataTableGroupBy/propTypes.js:3]: Looks like we are missing `options` here

**Review by jcfilben — COMMENTED** [review]
Just noticed the propTypes.js file is missing props. Everything else looks good

File: src/js/components/DataTableGroupBy/propTypes.js (hunk 2 above)
Comment (ericsoderberghp) [reply src/js/components/DataTableGroupBy/propTypes.js]: added propTypes

**Review by jcfilben — APPROVED** [review]
Looks good!

**Review by taysea — APPROVED** [review]
LGTM.

**Pull request merged**